presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -437,13 +438,18 @@ private PrestoSparkTaskSourceRdd createTaskSourcesRdd(
         int partitionCount = 0;
         boolean autoTunePartitionCount = isSparkPartitionCountAutoTuneEnabled(session);
         if (autoTunePartitionCount) {
+            int minPartitionCount = getMinSparkInputPartitionCountForAutoTune(session);
             int maxPartitionCount = getMaxSparkInputPartitionCountForAutoTune(session);
+            checkArgument(minPartitionCount <= maxPartitionCount,
+                    "Min partition count for auto tune %s, cannot be larger than max partition count %s",
+                    minPartitionCount,
+                    maxPartitionCount);
 
             for (int splitIndex = 0; splitIndex < splits.size(); splitIndex++) {
                 int partitionId;
                 long splitSizeInBytes = splits.get(splitIndex).getSplit().getConnectorSplit().getSplitSizeInBytes().getAsLong();
 
-                if ((!queue.isEmpty() && queue.peek().getSplitsInBytes() + splitSizeInBytes <= maxSplitsSizeInBytesPerPartition)
+                if ((partitionCount >= minPartitionCount && queue.peek().getSplitsInBytes() + splitSizeInBytes <= maxSplitsSizeInBytesPerPartition)
```

Comment:
Note this is probably not enough. Thinking about the following case:

- `minPartitionCount` = 100
- 'maxSplitsSizeInBytesPerPartition` = 1GB
- total input size is 10GB. 

Will this still get you 100 input partitions? :)